src/components/ChemicalEditors.vue

```diff
@@ -22,6 +22,69 @@ export default {
   },
   props: {
     type: String
+  },
+  methods: {
+    save: function() {
+      // Determine which editor is being saved.
+      if (this.type === "definedCompound") {
+        //todo: this is not going to be implemented on this ticket
+      } else {
+        this.saveIllDefinedCompound();
+      }
+    },
+    saveIllDefinedCompound() {
+      let compoundId = this.$store.state.compound.illdefinedcompound.data.id;
+      // Generic pot body
+      let requestBody = {
+        type: "illDefinedCompound",
+        attributes: {
+          mrvfile: this.$refs["marvin"].mrvfile
```

Comment:
we have a reference to the mrvfile in both `this.$refs.marvin.mrvfile` as well as `this.mrvfile` from w/in the MarvinWindow component. Would it be better to mapState in the MarvinWindow component into the compound store w/ something like `editor_structure` that could be a computed property of either the marvin or ketcher components that we could use to build the `requestBody`?

this may not make sense, but it seems like using a $ref as a computed property is [frowned upon in vuejs](https://vuejs.org/v2/guide/components-edge-cases.html#Accessing-Child-Component-Instances-amp-Child-Elements)

> $refs are only populated after the component has been rendered, and they are not reactive. It is only meant as an escape hatch for direct child manipulation - you should avoid accessing $refs from within templates or computed properties.

[here](https://stackoverflow.com/questions/39035498/vuejs-watch-refs) is an SO post as well, it seems like it is working this way so it may make sense to go with it, but shouldn't we imagine that there will only be one structure string (mol/mrvfile) that is being edited at a time? Wouldn't we want to unbind all of the store attributes to compound when the user flips from marvin to ketcher? It's great to populate all this data into state onload but it begs the question of how to manage it through UX?

I'm not sure of what's best, could the `saveIllDefinedCompound` function just rely on what is in state to dispatch the necessary actions for the post/patch requirements? or do we modify those conditions dependent upon some data/computed props in the component and some from state? I would think it easier to just base it off of state alone, but that may require more API calls.

Reply:
> $refs are only populated after the component has been rendered, and they are not reactive. It is only meant as an escape hatch for direct child manipulation - you should avoid accessing $refs from within templates or computed properties.

Regarding this point,  I'm not using this in or as a watched or computed value (in fact there are no watched or computed values on this component) and I don't need reactivity from the mrvfile's value.  All this `$refs.marvin.mrvfile` does is pull whatever value is currently saved on the marvin component when the save button is clicked.  That value is then used to construct the body of the patch or post request.

If your feeling is that the MarvinWindow and KetcherWindow components should be responsible for saving their own compounds, then I'd be happy to implement it that way (looking at it now, that makes a lot of sense).  

> I'm not sure of what's best, could the saveIllDefinedCompound function just rely on what is in state to dispatch the necessary actions for the post/patch requirements? or do we modify those conditions dependent upon some data/computed props in the component and some from state? I would think it easier to just base it off of state alone, but that may require more API calls.

I'm opposed to having the MarvinWindow component update the store outside POSTing/PATCHing.  The local mrvfile on the MarvinWindow contains all potential changes.  All of the methods on the MarvinWindow component are all tools to help us interact with the iframe.

[Vuex's documentation](https://vuex.vuejs.org/guide/structure.html) weighs in on this too:


```
Vuex doesn't really restrict how you structure your code. Rather, it enforces a set of high-level principles
1. Application-level state is centralized in the store.
2. The only way to mutate the state is by committing mutations, which are synchronous transactions.
3. Asynchronous logic should be encapsulated in, and can be composed with actions.
```

I believe having any component's changes be kept track of using the store would be a violation of their core principles 1 and 2.

If you were to push for having one "state" manage this page.  I feel the compound information would need to be loaded and tracked local to the MarvinWindow and KetcherWindow components.  This would require a rewrite to the search bar on the navbar and require the Vuex separation api from the Vuex actions in the docs linked above.

**TL;DR:**  
We should leave the ref as is or move the ill defined compound save onto the Marvin window, which I am in favor of doing.
Unless we are going to implement large structural changes that might make things more difficult, a local and Vuex state are required for managing the Marvin and Ketcher windows (as well as many others).  I am in favor of changing how we do this, but we should at least do a Monday tech talk on it.